src/tests.rs

```diff
@@ -1289,3 +1289,41 @@ fn parse_mod() {
     let mod_decl = parse_declaration_checked(expr);
     assert_debug_snapshot!(mod_decl);
 }
+
+// ==================
```

Comment:
I would also add a test for something like:

```rust
#[cfg(FALSE)]
trait Foo {
    type X = i32;
    const N: i32 = 42;
}
```

Eg a trait declaration which is semantically invalid but syntactically valid.

Reply:
Don't we already test those, with `SHORT_NAME` and `TypeWithDefault`?
```rs
const LONG_NAME: &'static str;
const SHORT_NAME: char = 'T';

type AssocType: Bound;
type TypeWithDefault = Rc<RefCell<MyStruct>>;
```

Reply:
It might be more expressive to add a test where those are in a trait body, to show the range of accepted syntax?

Eh, it doesn't really matter. The types are orthogonal enough that you can probably assume that if some construct parses in one context, it parses in another.